core/controlplane/config/templates/stack-template.json

```diff
@@ -895,6 +901,50 @@
       }
     },
     {{ end -}}
+    {{ if .LoadBalancer.LoadBalancerV2 }}
+    "{{.LoadBalancer.LogicalName}}TargetGroup": {
+      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
+      "Properties": {
+        "HealthCheckIntervalSeconds": "10",
+        "HealthyThresholdCount": "3",
```

Comment:
it seems to be spamming APIserver with non-SSL healthchecks which produce annoying entry in logs, does `HealthCheckProtocol: HTTPS` work?

Reply:
TCP is the only supported protocol for NLB health checks at this point.

Reply:
es, I just checked the API server logs, and there are a lot of TLS handshake errors from these health checks. Do you have a suggestion on how can we address this?

Reply:
it seems there is no way out of it without patching apiserver to not to complain about it on info leveld.

Reply:
Just FYI, there are HTTP and HTTPS health checks on NLB ([docs](http://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html))

Reply:
I don't see this here:

![image](https://user-images.githubusercontent.com/29534/32407636-1780e870-c173-11e7-9bd3-2ae176c3749b.png)

As far as I know, HTTP(S) health checks are only available for application load balancers (ALB, L7); NLBs operate on the TCP layer.

Reply:
Once you create an NLB listener and HealthCheck, the AWS console doesn't let you change it to HTTP/HTTPS (unfortunately). Try (manually) making a new Target Group with an HTTPS health check on an alternate port. You can then attach it to the NLB. I'm doing it right here: https://github.com/micahhausler/kubernetes/blob/662f4eee0d11d07bc2d0ca64198f2b0c051f684e/pkg/cloudprovider/providers/aws/aws_loadbalancer.go#L499-L514

Reply:
To be clear, the listener protocol is TCP, but the _target group_ health check can be HTTPS

Reply:
Great, thanks for letting us know! That's great news.